src/document.cpp

```diff
@@ -265,19 +265,19 @@ rvoe<NoReturnValue> PdfDocument::init() {
     switch(opts.output_colorspace) {
     case CAPY_DEVICE_CS_RGB:
         if(!cm.get_rgb().empty()) {
-            output_profile = store_icc_profile(cm.get_rgb(), 3);
+            output_profile = *add_icc_profile(cm.get_rgb(), 3);
         }
         break;
     case CAPY_DEVICE_CS_GRAY:
         if(!cm.get_gray().empty()) {
-            output_profile = store_icc_profile(cm.get_gray(), 1);
+            output_profile = *add_icc_profile(cm.get_gray(), 1);
         }
         break;
     case CAPY_DEVICE_CS_CMYK:
         if(cm.get_cmyk().empty()) {
             RETERR(OutputProfileMissing);
         }
-        output_profile = store_icc_profile(cm.get_cmyk(), 4);
+        output_profile = *add_icc_profile(cm.get_cmyk(), 4);
```

Comment:
All these dereferences need to be converted to propagate with `ERC`. Otherwise it will segfault in case any errors happen.

Reply:
I'm not sure what that means, is that just using `ERC(add_icc_profile(...))` here?

Reply:
`ERC` is a macro for "propagate error upwards if it exists". Kind of like how Rust's error propagation works. I did it like this both just to experiment how `std::expected` works and to make the code work even if exceptions and rtti are disabled.

It ended up a bit unergonomic, but in this case basically:

```
ERC(retval, add_icc_profile(cm.get_cmyk(), 4);
output_profile = retval;
```

Reply:
OK that should fix that.